t/unit/backends/test_database.py

```diff
@@ -862,8 +862,8 @@ def test_on_chord_part_return_failure(self):
         request.id = tid2
         self.b.mark_as_failure(tid2, ValueError(), request=request)
 
-        with pytest.raises(ChordCounter.DoesNotExist):
-            ChordCounter.objects.get(group_id=gid)
+        chord_counter.refresh_from_db()
+        assert chord_counter.count == 1
```

Comment:
I don't think this is correct, we have a header of 2, 1 is done, the other fails.
If a task in the header fails the chord is done and the callback is not called.

If this is the case, a ChordCounter object will remain in the db for every failed chord.